src/consensus/aft/raft.h

```diff
@@ -292,7 +316,17 @@ namespace aft
 
       // This will not work once we have reconfiguration support
       // https://github.com/microsoft/CCF/issues/1852
-      auto active_nodes_ = active_nodes();
+      auto active_nodes_ = active_node_ids();
+      if (!catching_up && !retiring)
+      {
+        active_nodes_.insert(id());
+      }
+#ifndef NDEBUG
+      std::stringstream ss;
+      for (auto& m : active_nodes_)
+        ss << " " << m;
+      LOG_DEBUG_FMT("Active nodes:{}", ss.str());
+#endif
```

Comment:
You can build this concatenation with `fmt::join`. Something like

```suggestion
      LOG_DEBUG_FMT("Active nodes: {}", fmt::join(active_nodes_, ", "));
```